src/MongoDB/ClientEncryption.c

```diff
@@ -943,6 +1089,10 @@ static void phongo_clientencryption_decrypt(php_phongo_clientencryption_t* clien
 
 	php_phongo_zval_to_bson_value(zciphertext, PHONGO_BSON_NONE, &ciphertext);
 
+	if (EG(exception)) {
+		goto cleanup;
```

Comment:
This is added for consistency but we don't actually test for this since `decrypt()` expects a Binary and that is always going to successfully encode to BSON.

This is also mentioned in the commit message.

Reply:
You mean it's inaccessible code, intentionally added? That seems strange to me. A comment in the code would be more appropriate.

Reply:
See https://github.com/mongodb/mongo-php-driver/pull/1412#discussion_r1206321486 for a recent discussion about this very issue in @alcaeus' PR.

I think ideally we'd use assertions here, as is sometimes done in libmongoc; however, there's no prior art in PHPC for doing so.

I created [PHPC-2235](https://jira.mongodb.org/browse/PHPC-2235) to track this, but I'd prefer to leave this in place for now since it'd be easily detected by tooling should we end up integrating that.